src/transformers/utils/quantizers.py

```diff
@@ -0,0 +1,879 @@
+                " quantization docs: https://huggingface.co/docs/transformers/main_classes/quantization "
+            )
+
+        return kwargs
+
+    def get_quantizer(self, config: PretrainedConfig) -> "HFQuantizer":
+        """
+        Detect quantization type from model config, resolve conflicts with config from args.
+
+        Returns:
+            Quantizer instance to use for model loading.
+        """
+        # set starting values based on args ========================================================================
+        if not hasattr(self, "quantization_method"):
+            raise AttributeError("quantization_config from args must be parsed before parsing model config")
+        quantization_status = QuantizationStatus.FRESH if self.quantization_config is not None else None
+
+        # parse model coonfig and handle conflicts ==================================================================
+        if hasattr(config, "quantization_config"):
+            quantization_method_from_config = config.quantization_config.get(
+                "quant_method", QuantizationMethod.BITS_AND_BYTES
+            )
+
+            if self.quantization_method is not None and quantization_status == QuantizationStatus.FRESH:
+                if quantization_method_from_config != QuantizationMethod.GPTQ:
+                    logger.warning(
+                        "You passed `quantization_config` or equivalent parameters to `from_pretrained` but the model you're loading already has a"
+                        " `quantization_config` attribute. The `quantization_config` from the model will be prevail."
+                    )
+                    self.quantization_method = quantization_method_from_config  # This is differnt from previous code where from_args prevailed
+                    quantization_status = QuantizationStatus.PREQUANTIZED
+                else:
+                    # special case for GPTQ config collision   # TODO consider unification
+                    loading_attr_dict = self.quantization_config.get_loading_attributes()
+                    for attr, val in loading_attr_dict.items():
+                        config.quantization_config[attr] = val
+
+                    self.quantization_config = config.quantization_config
+                    self.quantization_method = quantization_method_from_config
+                    logger.warning(
+                        "You passed `quantization_config` to `from_pretrained` but the model you're loading already has a "
+                        "`quantization_config` attribute and has already quantized weights. However, loading attributes"
+                        " (e.g. disable_exllama, use_cuda_fp16, max_input_length) will be overwritten with the one you passed to `from_pretrained`. "
+                        "The rest will be ignored."
+                    )
+                    quantization_status = QuantizationStatus.PREQUANTIZED
```

Comment:
simplified a bit. pls check

is it correct that you want get_loading_attributes always have priority over config.quantization_config?

Reply:
Yes, `get_loading_attributes` is used to modify `config.quantization_config` so that we can for example change the default kernel (gptq) or use fusion (awq).

Reply:
get_loading_attributes take precedence in this PR too, with warning.
see `QuantizationConfigParser.handle_configs_collision()`